test/test_chat.py

```diff
@@ -1,15 +1,41 @@
-import os
+import sys
+import unittest
 
-class TestChat:
+from wp_parser.wp_chat import main as parser
+
+# for capturing print() output
+from contextlib import contextmanager
+try:
+    from StringIO import StringIO
+except ImportError: #python3
+    from io import StringIO
+
+
+@contextmanager
+def captured_output():
+    new_out, new_err = StringIO(), StringIO()
+    old_out, old_err = sys.stdout, sys.stderr
+    try:
+        sys.stdout, sys.stderr = new_out, new_err
+        yield sys.stdout, sys.stderr
+    finally:
+        sys.stdout, sys.stderr = old_out, old_err
+
+
+class TestChat(unittest.TestCase):
+    def test_chat_1(self):
+        # out_filename = str(tmpdir.join("abc"))
+        # out_filename = ""
 
-    def test_chat(self, tmpdir):
-        out_filename = str(tmpdir.join("abc"))
         for case in ['One', 'Two']:
-            cmd = 'python -p wp_parser/wp_chat.py -f test/testChat2.txt -n Username{} > {}'.format(case, out_filename)
-            os.system(cmd)
-            with open(out_filename) as fh:
-                result = fh.read()
+            # creating mock args
+            args = "-f test/testChat2.txt --root Username{}".format(case).split()
+
+            with captured_output() as (result, err):
+                parser(args)
```

Comment:
rather than using ``os.system``, better approach is to import the main package and pass args directly to it.